packages/mobile/src/web3/contracts.ts

```diff
@@ -4,30 +4,49 @@
  * Leaving the name for recognizability to current devs
  */
 import { ContractKit, newKitFromWeb3 } from '@celo/contractkit'
-import { RpcWallet } from '@celo/contractkit/lib/wallets/rpc-wallet'
 import { sleep } from '@celo/utils/src/async'
 import { call, delay, select } from 'redux-saga/effects'
 import { ContractKitEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
 import { ErrorMessages } from 'src/app/ErrorMessages'
 import { DEFAULT_FORNO_URL } from 'src/config'
-import { isProviderConnectionError } from 'src/geth/geth'
+import { getGethInstance, isProviderConnectionError } from 'src/geth/geth'
 import { waitForGethInitialized } from 'src/geth/saga'
 import { navigateToError } from 'src/navigator/NavigationService'
 import Logger from 'src/utils/Logger'
 import { getHttpProvider, getIpcProvider } from 'src/web3/providers'
 import { fornoSelector } from 'src/web3/selectors'
 import Web3 from 'web3'
-import { IpcProvider } from 'web3-core'
+import { RNGethWallet } from '../geth/RNGethWallet'
 
 const TAG = 'web3/contracts'
 const KIT_INIT_RETRY_DELAY = 2000
 const CONTRACT_KIT_RETRIES = 3
 
-let ipcProvider: IpcProvider | undefined
-let gethWallet: RpcWallet | undefined
+let gethWallet: RNGethWallet | undefined
 let contractKit: ContractKit | undefined
 
+function* initWallet() {
```

Comment:
these changes seem well-intentioned and positive but I have struggled a lot in the past in this file
have you tested these changes yet?

Reply:
Not yet because I had some uncertainty around how to change some of the geth related init sagas to better fit new needs and I didn't want to spend too much time on testing before I got the chance to talk with some people about this to see if the direction is ok. 

But I had a productive chat with @annakaz today, so I'll be doing some refactoring here an will update+add tests as well for when I'm asking for the final PR.

Maybe I didn't make this clear but I didn't consider this final in any way, I shared the PRs because I wanted feedback before polishing the solution.